### PR TITLE
feat: apply new color design for switch - ACC-138

### PR DESCRIPTION
**Pull request by BrzozowskiPawel**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/ACC-138" title="ACC-138" target="_blank">StoryACC-138</a>  Contrast distance from text to background (1.4.3)
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Updated the color of the switch and also added a border to match Figma design.
I don't know why but tests (snapshots) seem to do not see the change. 

### Issues

After the chat with the designers, some colors have been changed in the design so it needed to be updated in the code aswell.

### Solutions

Added border for specific appearance and also changed the color of background.

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by agisilaos — APPROVED** [review]

File: Wire-iOS/Sources/UserInterface/UserProfile/Devices/ProfileClientViewController.swift [hunk 1]
```diff
@@ -343,6 +343,7 @@ final class ProfileClientViewController: UIViewController, SpinnerCapable {
 
     @objc
     private func onTrustChanged(_ sender: AnyObject) {
+
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/UserProfile/Devices/ProfileClientViewController.swift:346]:
```suggestion
```

File: Wire-iOS/Sources/UserInterface/Conversation/Create/ToggleView.swift [hunk 2]
```diff
@@ -1,88 +0,0 @@
...
-// GNU General Public License for more details.
-//
-// You should have received a copy of the GNU General Public License
-// along with this program. If not, see http://www.gnu.org/licenses/.
-//
-
-import UIKit
-import WireCommonComponents
-
-final class ToggleView: UIView, Themeable {
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Conversation/Create/ToggleView.swift:22]: Was it dead code?

File: Wire-iOS/Sources/Components/Styles/SwitchStyle.swift [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+    let switchStyle: SwitchStyle
+
+    override var isOn: Bool {
+        didSet {
+            guard isOn != oldValue else { return }
+            valueDidChange()
+        }
+    }
+
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/Components/Styles/SwitchStyle.swift:47]:
```suggestion
// MARK: - Initialization
```

File: Wire-iOS/Sources/Components/Styles/SwitchStyle.swift [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    static let `default` = SwitchStyle(
+        enabledOnStateColor: SemanticColors.SwitchColors.backgroundSwitchOnStateEnabled,
+        enabledOffStateColor: SemanticColors.SwitchColors.backgroundSwitchOffStateEnabled,
+        enabledOnStateBorderColor: SemanticColors.SwitchColors.borderSwitchOnStateEnabled,
+        enabledOffStateBorderColor: SemanticColors.SwitchColors.borderSwitchOffStateEnabled
+    )
+}
+
+class Switch: UISwitch, Stylable {
+
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/Components/Styles/SwitchStyle.swift:38]:
```suggestion

// MARK: - Properties

```

File: Wire-iOS/Sources/Components/Styles/SwitchStyle.swift [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    static let `default` = SwitchStyle(
+        enabledOnStateColor: SemanticColors.SwitchColors.backgroundSwitchOnStateEnabled,
+        enabledOffStateColor: SemanticColors.SwitchColors.backgroundSwitchOffStateEnabled,
+        enabledOnStateBorderColor: SemanticColors.SwitchColors.borderSwitchOnStateEnabled,
+        enabledOffStateBorderColor: SemanticColors.SwitchColors.borderSwitchOffStateEnabled
+    )
+}
+
+class Switch: UISwitch, Stylable {
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/Components/Styles/SwitchStyle.swift:37]:
```suggestion
final class Switch: UISwitch, Stylable {
```

File: Wire-iOS/Sources/Components/Styles/SwitchStyle.swift [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        super.init(frame: .zero)
+        applyStyle(switchStyle)
+
+        addTarget(self, action: #selector(valueDidChange), for: .valueChanged)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/Components/Styles/SwitchStyle.swift:59]:
```suggestion

// MARK: - Methods

```

File: Wire-iOS/Sources/Components/Styles/SwitchStyle.swift [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    @objc
+    private func valueDidChange() {
+        applyStyle(switchStyle)
+    }
+
+    public func applyStyle(_ style: SwitchStyle) {
+        backgroundColor = style.enabledOffStateColor
+        onTintColor = style.enabledOnStateColor
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/Components/Styles/SwitchStyle.swift:67]:
```suggestion
        onTintColor = style.enabledOnStateColor

```

File: Wire-iOS/Sources/UserInterface/Conversation/Create/ToggleView.swift (hunk 2 above)
Comment (agisilaos) [reply Wire-iOS/Sources/UserInterface/Conversation/Create/ToggleView.swift:22]: @KaterinaWire Yes!

File: Wire-iOS/Sources/Components/Styles/SwitchStyle.swift (hunk 7 above)
Comment (KaterinaWire) [reply Wire-iOS/Sources/Components/Styles/SwitchStyle.swift:67]: @BrzozowskiPawel This suggestion too

File: Wire-iOS/Sources/Components/Styles/SwitchStyle.swift [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+    private(set) var enabledOnStateBorderColor: UIColor
+    private(set) var enabledOffStateBorderColor: UIColor
+    private(set) var borderWidth: CGFloat = 1
+
+    static let `default` = SwitchStyle(
+        enabledOnStateColor: SemanticColors.Switch.backgroundOnStateEnabled,
+        enabledOffStateColor: SemanticColors.Switch.backgroundOffStateEnabled,
+        enabledOnStateBorderColor: SemanticColors.Switch.borderOnStateEnabled,
+        enabledOffStateBorderColor: SemanticColors.Switch.borderOffStateEnabled
+    )
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/Components/Styles/SwitchStyle.swift:34]:
```suggestion
    )

```

File: Wire-iOS/Sources/Components/Styles/SwitchStyle.swift (hunk 7 above)
Comment (BrzozowskiPawel) [reply Wire-iOS/Sources/Components/Styles/SwitchStyle.swift:67]: This causes a swiftlint error. I can code it but it will gives us back an error

**Comment by BrzozowskiPawel** [comment]
Should be all good now with suggestions implemented :)

**Review by KaterinaWire — APPROVED** [review]

**Pull request merged**